examples/gno.land/r/demo/disperse/disperse.gno

```diff
@@ -0,0 +1,108 @@
+package disperse
+
+import (
+	"std"
+
+	tokens "gno.land/r/demo/grc20factory"
+)
+
+// Get address of Disperse realm
+var realmAddr = std.CurrentRealm().Addr()
+
+// DisperseUgnot parses receivers and amounts and sends out ugnot
+// The function will send out the coins to the addresses and return the leftover coins to the caller
+// if there are any to return
+func DisperseUgnot(addresses []std.Address, coins std.Coins) {
+	coinSent := std.GetOrigSend()
+	caller := std.PrevRealm().Addr()
+	banker := std.GetBanker(std.BankerTypeOrigSend)
+
+	// Check if realm balance is equal to amount sent
+	if banker.GetCoins(realmAddr).AmountOf("ugnot") != coinSent.AmountOf("ugnot") {
+		panic(ErrBalanceNotEqToAmountSent)
+	}
+
+	if len(addresses) != len(coins) {
+		panic(ErrNumAddrValMismatch)
+	}
+
+	for _, coin := range coins {
+		if coin.Amount <= 0 {
+			panic(ErrNegativeCoinAmount)
+		}
+	}
+
+	var totalAmount int64
+	for _, coin := range coins {
+		totalAmount += coin.Amount
+	}
+
+	// Send coins
+	for i, _ := range addresses {
+		banker.SendCoins(realmAddr, addresses[i], std.NewCoins(coins[i]))
+	}
+
+	// Return possible leftover coins
+	for _, coin := range coinSent {
+		leftoverAmt := banker.GetCoins(realmAddr).AmountOf(coin.Denom)
+		if leftoverAmt > 0 {
+			send := std.Coins{{coin.Denom, leftoverAmt}}
+			banker.SendCoins(realmAddr, caller, send)
+		}
+	}
+}
+
+// DisperseGRC20 disperses tokens to multiple addresses
+// Note that it is necessary to approve the realm to spend the tokens before calling this function
+// see the corresponding filetests for examples
+func DisperseGRC20(addresses []std.Address, amounts []uint64, symbols []string) {
+	caller := std.PrevRealm().Addr()
+
+	if (len(addresses) != len(amounts)) || (len(amounts) != len(symbols)) {
+		panic(ErrArgLenAndSentLenMismatch)
+	}
+
+	for i := 0; i < len(addresses); i++ {
+		// Transfer tokens into the realm
+		tokens.TransferFrom(symbols[i], caller, std.CurrentRealm().Addr(), amounts[i])
```

Comment:
yeah I thought so, we kept it because it was done in the original implementation in solidity :
```sol
function disperseToken(IERC20 token, address[] recipients, uint256[] values) external {
        uint256 total = 0;
        for (uint256 i = 0; i < recipients.length; i++)
            total += values[i];
        require(token.transferFrom(msg.sender, address(this), total));
        for (i = 0; i < recipients.length; i++)
            require(token.transfer(recipients[i], values[i]));
    }
```
is this also the case for ethereum ? or am I missing something that differs between the implementations themselves ?

as for the fix :
https://github.com/gnolang/gno/pull/2613/commits/d05632a7711b9629469fc981cf295631ac0a9a57

Reply:
Hmm, maybe it's due to an old version of ERC20. From what I saw with the current implementation, `transferFrom` in Solidity ([OZ](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/0b58a783b9b33b63eef2994af8958c0c6a72dc51/contracts/token/ERC20/ERC20.sol#L190)) reverts on insufficient balance, but that might not be the case for the 0.4.X version which Disperse is made with.

If the tests pass, lets keep it only with `transferFrom` 👍